PR [2pt]: Longitudinal Adjustment in the Surface Area

hhs732 opened the pull request:
This PR aims to longitudinally refine the discharge values in the rating curve by filtering the surface area values and recalculating discharge and the rest of Manning equations' variables, including bed area and volume. You can find the details of the framework here:

https://docs.google.com/presentation/d/1rbOWHimHTTb9xpMBdSRu1YLxgDw21GNmXb9OE5tkquI/edit?slide=id.g10245655793_2_0#slide=id.g10245655793_2_0

### Changes

- /src/
    - longitudinal_flow_adjudtment.py
    - add_crosswalk.py
    - bash_valiables.env

Closes #1676 
Closes #1565 
---------------------------------------------------------------
### Testing
This PR has been tested on 50 HUC8 across the country. 

---------------------------------------------------------------
### Deployment Plan (For FIM developers use)
- **Does the change impact inputs, docker or python packages?**
    - [x] Yes
    - [ ] No  (f no.. skip the rest of the Deployment Plan section)
    
-  **If you are not a FIM dev team member:  Please let us know what you need and we can help with it.**

-  **If you are a FIM Dev team member:** 
    -  Please work with the DevOps team and do not just go ahead and do it without some co-ordination.
    -  Copy where you can, assign where you can not, and it is your responsibility to ensure it is done. Please ensure it is completed before the PR is merged. 
    
    - Has new or updated python packages, PipFile, Pipefile.lock or Dockerfile changes?  DevOps can help or take care of it if you want. Just need to know if it is required.
       - [ ] Yes
       - [x] No
    - Require new or adjusted data inputs? Does it have a way to version (folder or file dates)?
       - [x] No
       - [ ] Yes
           -  Require new pre-clip set or any other data reloads, such as DEMS, osm, etc. ie.. pre-requisite re-data upstream of your input changes.
                - [ ] Yes
                - [x] No
           -  Has the inputs been copied/exist in all four enviros:
                 - [ ] FIM EFS  
                 - [ ] FIM S3
                 - [ ] ESIP
                 - [ ] Dev1
            
- Please use caution in removing older version unless it is at least two versions ago.  Confirm with DevOps if cleanup might be involved.

- If new or updated data sets, has the FIM code, including running fim_pipeline.sh, been updated and tested with the new/adjusted data? You can dev test against subsets if you like.
    - [N/A] Yes

### Notes to DevOps Team or others:
Please add any notes that are helpful for us to make sure it is all done correctly. Do not put actual server names or full true paths, just shortcut paths like 'efs..../inputs/,  or 'dev1....inputs', etc.



---------------------------------------------------------------
### Issuer Checklist (For developer use)

You may update this checklist before and/or after creating the PR. If you're unsure about any of them, please ask, we're here to help! These items are what we are going to look for before merging your code.

- [x] Informative and human-readable title, using the format: `[_pt] PR: <description>`
- [x] Links are provided if this PR resolves an issue, or depends on another other PR
- [x] If submitting a PR to the `dev` branch (the default branch), you have a descriptive [Feature Branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) name using the format: `dev-<description-of-change>` (e.g. `dev-revise-levee-masking`)
- [x] Changes are limited to a single goal (no scope creep)
- [x] The feature branch you're submitting as a PR is up to date (merged) with the latest `dev` branch
- [x] `pre-commit` hooks were run locally
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] [CHANGELOG](/docs/CHANGELOG.md) updated with template version number, e.g. `4.x.x.x`
- [x] Add yourself as an [assignee](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) in the PR  as well as the FIM Technical Lead

---------------------------------------------------------------
### Merge Checklist (For Technical Lead use only)

- [ ] Update [CHANGELOG](/docs/CHANGELOG.md) with latest version number and merge date
- [ ] Update the [Citation.cff](/CITATION.cff) file to reflect the latest version number in the [CHANGELOG](/docs/CHANGELOG.md)
- [ ] If applicable, update [README](/README.md) with major alterations

hhs732 committed:
Redefining longitudinal filtering based on SurfaceArea and Volume recalculation

hhs732 committed:
Changing the slope for select feature-ids in the greater Pittsburgh area

hhs732 committed:
Updating slope and manning roughness for the greater Pittsburgh area

hhs732 committed:
Merge with dev

hhs732 committed:
modified:   src/add_crosswalk.py

hhs732 committed:
Merge branch 'dev' of https://github.com/NOAA-OWP/inundation-mapping into dev-longitudinal-adjustment-surfacearea-base-WQ-volcal0SAhIDs

hhs732 committed:
Thalwag adjustmen warning added

RyanSpies-NOAA reviewed (dismissed):
Reviewed code and tested on three HUC8s. Confirmed there were no errors and qualitative review of inundation extent shows improvement over current FIM6 inundation.

hhs732 committed:
Update vmann_input_file path in bash_variables.env

hhs732 committed:
Updated CHANGELOG

Added details for version 4.x.x.x release, including changes to discharge values and Manning equations.

RobHanna-NOAA committed:
Update CHANGELOG.md

RobHanna-NOAA committed:
Update CHANGELOG.md
